XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/Simulation/Hands/SimulatedHandControllerPose.cs

```diff
@@ -0,0 +1,289 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using System;
+using System.Collections.Generic;
+using XRTK.Definitions.Utilities;
+using XRTK.Services;
+using XRTK.Definitions.Controllers.Simulation.Hands;
+using XRTK.Providers.Controllers.Hands;
+
+namespace XRTK.Providers.Controllers.Simulation.Hands
+{
+    /// <summary>
+    /// Simulatd pose of an hand controller defined by recorded joint poses.
+    /// </summary>
+    public class SimulatedHandControllerPose
+    {
+        private static readonly int jointCount = Enum.GetNames(typeof(TrackedHandJoint)).Length;
+        private static readonly Dictionary<string, SimulatedHandControllerPose> handPoses = new Dictionary<string, SimulatedHandControllerPose>();
+        private static bool isInitialized = false;
+
+        /// <summary>
+        /// Joint poses are stored as right-hand poses in camera space.
+        /// Output poses are computed in world space, and mirroring on the x axis for the left hand.
+        /// </summary>
+        private MixedRealityPose[] localJointPoses;
```

Comment:
Why does a pose have an array of poses?

Reply:
The hand pose is made up of individual joint poses. Makes sense?

Reply:
Unrelated note - iirc MixedRealityPose is a class in MRTK. It could be a struct, as I did in my codebase. Less garbage will be generated if it is a struct, since each frame there are tons of init calls for them.

Reply:
`MixedRealityPose` is a struct in XRTK. Let me see if I can make `SimulatedHandControllerPose` a struct. The naming is somewhat confusing here it seems. I might change that.